packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+    }
+  }
+}
+
+/**
+ * Response to return in `run` function.
+ */
+class RunJsonResponse {
+  /**
+   *
+   * @param {!bool} success Whether the command was successful.
+   * @param {?string=} errorMessage Defaults to null.
+   * @param {?DebugResult=} debugResult Curated results from Xcode's debug
+   *     function. Defaults to null.
+   */
+  constructor(success, errorMessage = null, debugResult = null) {
+    this.status = success;
+    this.errorMessage = errorMessage;
+    this.debugResult = debugResult;
+  }
+
+  /**
+   * Converts this object to a JSON string.
+   *
+   * @returns {!string}
+   * @throws Throws an error if conversion fails.
+   */
+  stringify() {
+    return JSON.stringify(this);
+  }
+}
+
+/**
+ * Utility class to return a result along with a potential error.
+ */
+class FunctionResult {
+  /**
+   *
+   * @param {?Object} result
+   * @param {?string=} error Defaults to null.
+   */
+  constructor(result, error = null) {
+    this.result = result;
+    this.error = error;
+  }
+}
+
+/**
+ * Curated results from Xcode's debug function. Mirrors parts of
+ * `scheme action result` from Xcode's Script Editor dictionary.
+ */
+class DebugResult {
+  /**
+   *
+   * @param {!Object} result
+   */
+  constructor(result) {
+    this.completed = result.completed();
+    this.status = result.status();
+    this.errorMessage = result.errorMessage();
+  }
+}
+
+/**
+ * Get the Xcode application from the given path. Since macs can have multiple
+ * Xcode version, we use the path to target the specific Xcode application.
+ * If the Xcode app is not running, return null with an error.
+ *
+ * @param {!CommandArguments} args
+ * @returns {!FunctionResult} Return either an `Application` (Mac Scripting class)
+ *     or null as the `result`.
+ */
+function getXcode(args) {
+  try {
+    const xcode = Application(args.xcodePath);
+    const isXcodeRunning = xcode.running();
+
+    if (isXcodeRunning === false) {
+      return new FunctionResult(null, 'Xcode is not running');
+    }
+
+    return new FunctionResult(xcode);
+  } catch (e) {
+    return new FunctionResult(null, `Failed to get Xcode application: ${e}`);
+  }
+}
+
+/**
+ * After setting the active run destination to the targeted device, uses Xcode
+ * debug function from Mac Scripting for Xcode to install the app on the device
+ * and start a debugging session using the 'run' or 'run without building' scheme
+ * action (depending on `args.skipBuilding`). Waits for the debugging session
+ * to start running.
+ *
+ * @param {!Application} xcode An `Application` (Mac Scripting class) for Xcode.
+ * @param {!CommandArguments} args
+ * @returns {!FunctionResult} Return either a `DebugResult` or null as the `result`.
+ */
+function debugApp(xcode, args) {
+  const workspaceResult = waitForWorkspaceToLoad(xcode, args);
+  if (workspaceResult.error != null) {
+    return new FunctionResult(null, workspaceResult.error);
+  }
+	const targetWorkspace = workspaceResult.result;
```

Comment:
replace spaces for tab

```suggestion
  const targetWorkspace = workspaceResult.result;
```